chain/chain/src/chain.rs

```diff
@@ -1651,7 +1651,22 @@ impl Chain {
         let header = self.get_block_header(&sync_hash)?;
         let hash = *header.prev_hash();
         let prev_block = self.get_block(&hash)?;
-        let new_tail = prev_block.header().height();
+
+        // The congestion control added a dependency on the prev block when
+        // applying chunks in a block. This means that we need to keep the
+        // blocks at sync hash, prev hash and prev prev hash. The heigh of that
```

Comment:
```suggestion
        // blocks at sync hash, prev hash and prev prev hash. Height of that
```

You may also mention that `prev_block.height() + 1 == sync_block.height()` due to epoch finalisation restrictions, if it's not mentioned anywhere.